src/api/mutations/payment/usePostPaymentOptionMutation.ts

```diff
@@ -0,0 +1,34 @@
+import toast from 'react-hot-toast';
+import { useNavigate } from 'react-router-dom';
+import { useMutation } from '@tanstack/react-query';
+import httpClient from '@/api/httpClient';
+import { ENDPOINTS } from '@/config/endpoints.config';
+import { LOCAL_STORAGE_KEYS } from '@/config/local-storage.config';
+import { ROUTES } from '@/config/routes.config';
+import { ResponsePayment } from '@/types/payment.types';
+import { removeFromLocalStorage, setValueToLocalStorage } from '@/utils';
+
+function usePostPaymentOptionMutation(bookingId: string, paymentOption: string) {
+  const navigate = useNavigate();
```

Comment:
food for thought, no urgent need to fix that

your mutation has a lot of side effects (navigation, set value to local storage). 
It makes it difficult to reuse this mutation in other places of the app.

there is a [single responsibility principle](https://en.wikipedia.org/wiki/Single_responsibility_principle)

if we stuck to this principle, we would have only mutation-related code and the way to call side effects.
For this purpose, extra arguments can be passed to the mutation such as `onCompleted`, `onError` that can implement needed side effects